OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -870,7 +954,16 @@ void TryToRushAttack()
 
 				if (AttackOrFleeFuzzy.Rush.CanAttack(ownUnits, enemies))
 				{
-					var target = enemies.Any() ? enemies.Random(Random) : b;
+					Actor target = null;
+					for (;;)
+					{
+						// Ignore cloaked units...
+						// Can't handle them with detectors yet.
+						target = enemies.Any() ? enemies.Random(Random) : b;
+						if (!target.TraitsImplementing<Cloak>().Any())
```

Comment:
Please don't do this. Yes, I'm aware why are you doing this, but at least check if the trait is active, because this probably breaks TS due to the Stealth Generator.